### ISSUE TITLE
Code freezes when attempted to run on multiple GPUs

### ISSUE THREAD

**Issue by ashwinRameshBabu** [NVIDIA/DALI#946]
Hello all

I was trying to run the SUPERRES code in with pytorch.  The training works fine for 1 GPU(Defined in WS=1).  When I try to run it across multiple GPUs(WS = n), the code freezes.  Any help on how to fix this?

**Comment by JanuszL** [comment]
@Kh4L ?

**Comment by JanuszL** [comment]
Hi @ashwinRameshBabu,
Our example is based on what is available in [NVVL repository](https://github.com/NVIDIA/nvvl/blob/master/examples/pytorch_superres/run_distributed.sh), as I see setting `WS` is not sufficient, you need launch multiple processes with the appropriate `--rank` argument. Please check if that way works for you.
In your case, I guess the main process waits for other instances during sync while they don't exist.

**Comment by ashwinRameshBabu** [comment]
Hi @JanuszL 

Thanks for your response.  I am trying to run on a machine with 8 GPUs.  Other than WS what else need to be considered?  The --rank argument which is the base rank is set to 0 and the WS is set to 8.  Where am I missing?

What puzzles me is, I tried running the example provided in the NVVL repository, and the code works fine for multiple GPUs.  

output:
<img width="474" alt="issuePNG" src="https://user-images.githubusercontent.com/15481902/59053213-bb6b4080-8845-11e9-8b0e-2519838b8c8c.PNG">

**Comment by JanuszL** [comment]
I mean that you need to spawn multiple PyTorch processes in parallel. Please compare https://github.com/NVIDIA/DALI/blob/master/docs/examples/video/superres_pytorch/run.sh and https://github.com/NVIDIA/nvvl/blob/master/examples/pytorch_superres/run_distributed.sh.
We used NVVL as a base for our script, commented a few options and set LOADER to DALI.

**Comment by ashwinRameshBabu** [comment]
Thanks for your information.  One more question, 

I followed the method provided at nvvl's superres's run_distributed.sh to scale it to multiple GPUs.  I ran the code on 4 GPUs and I saw the following, [](url)
<img width="475" alt="s1" src="https://user-images.githubusercontent.com/15481902/59786834-055c1980-927d-11e9-8985-c92827d03ab2.PNG">

I can understand from the GPU processes that, for GPU: 1,2,3, part of GPU:0 is used for dataloading and augmentation.

Is there a way in DALI where i can use each GPU for its own dataloading and training?  I am triggering other GPUs the way it has been mentioned in the NVVl's example of superres which is, 

<img width="947" alt="s2" src="https://user-images.githubusercontent.com/15481902/59787047-73084580-927d-11e9-9c06-d4396f8e1c43.PNG">
